.github/workflows/go.yml

```diff
@@ -17,6 +17,7 @@ jobs:
       uses: actions/setup-go@v2
       with:
         go-version: ^1.16
+        fetch-depth: 0

```

Comment:
Supposedly it clones the whole history instead of just the latest revision. We need the history so that the linter can find out when issues were introduced and only complain about those introduced after the revision listed in our config. This definitely makes the linter slower btw, but it still runs in a few seconds so I think it is worth it to be able to enforce more rules. 

The call from the linter to `git` is still failing in CI though, which I'll come back to today. You're right that the reconciliation of what runs in our various jobs should be another PR, so I'll tackle that after.